sdks/keyvault/src/client.rs

```diff
@@ -173,7 +93,7 @@ impl<'a> KeyVaultClient<'a> {
             .get(&uri)
             .header(
                 "Authorization",
-                format!("Bearer {}", self.token.as_ref().unwrap().secret()),
+                format!("Bearer {}", self.token.as_ref().unwrap().token.secret()),
```

Comment:
May be a good time to switch to use [.bearer_auth](https://docs.rs/reqwest/0.10.8/reqwest/struct.RequestBuilder.html#method.bearer_auth). I'd also like to see `unwrap` replace with something that raises an error instead of panics.

Reply:
> 
> 
> May be a good time to switch to use [.bearer_auth](https://docs.rs/reqwest/0.10.8/reqwest/struct.RequestBuilder.html#method.bearer_auth). I'd also like to see `unwrap` replace with something that raises an error instead of panics.

For bearer_auth - sure, I'm updating the PR.

For unwrap: I'd make it a different PR and my suggestion is to have refresh_token return the token. As it is always called before the real request, the unwrap shall never fail as it mean the refresh_token returned Ok but token update was not done.